client/command/socks.go

```diff
@@ -0,0 +1,190 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/bishopfox/sliver/protobuf/sliverpb"
+
+	"github.com/bishopfox/sliver/client/core"
+	"github.com/bishopfox/sliver/protobuf/rpcpb"
+
+	"github.com/desertbit/grumble"
+)
+
+func socks(ctx *grumble.Context, rpc rpcpb.SliverRPCClient) {
+	session := ActiveSession.GetInteractive()
+	if session == nil {
+		return
+	}
+
+	port := ctx.Flags.Int("port")
+	go startSocksServer(ctx, port, rpc)
+}
+
+func startSocksServer(ctx *grumble.Context, port int, rpc rpcpb.SliverRPCClient) {
+	fmt.Printf(Info + fmt.Sprintf("Starting socks server on port %d\n\n", port))
+	session := ActiveSession.Get()
+	if session == nil {
+		fmt.Printf(Warn+"%s\n", "Current session is nil")
+		return
+	}
+
+	netAddr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("0.0.0.0:%d", port))
+	if err != nil {
+		fmt.Printf(Warn+"[0x07] %s\n", err)
+		return
+	}
+
+	listener, err := net.ListenTCP("tcp4", netAddr)
+	if err != nil {
+		fmt.Printf(Warn+"[0x08] %s\n", err)
+		return
+	}
+
+	var socksConnList []*net.TCPConn
+
+	// If session is closed then close the socks server
+	go func(implantSessionID uint32) {
+		// Continuously check if the connection to the implant that opened the socks server is still alive
+		for GetSession(fmt.Sprintf("%d", implantSessionID), rpc) != nil {
+		}
```

Comment:
I'm not fan of empty loops. We have an event system that you can subscribe to in order to know when a session is lost. I'd recommend using that instead of empty loops that would run infinitely.
Check out https://github.com/BishopFox/sliver/blob/master/client/console/console.go#L160-L169